matchmaking.py

```diff
@@ -146,6 +146,19 @@ def update_user_profile(self) -> None:
             except Exception:
                 pass
 
+    def get_weights(self, online_bots: list[USER_PROFILE_TYPE], rating_preference: str, min_rating: int, max_rating: int,
+                    game_type: str) -> list[int]:
+        """Get the weight for each bot. A higher weights means the bot is more likely to get challenged."""
+        weights = [1] * len(online_bots)
+        if rating_preference == "high":
+            reduce_ratings_by = min(min_rating - (max_rating - min_rating), min_rating - 1)

```

Comment:
Where does this calculation come from? Isn't `min_rating - (max_rating - min_rating)` just `max_rating`? Then `min` only really makes sense here in the case where `min_rating > max_rating`, which seems like it should never happen.

Maybe I'm missing something, but then a clarifying comment would be helpful for future readers.

Reply:
This calculation is the same as `2*min_rating-max_rating`. With this change, a bot with `max_rating` rating will be twice as likely to get picked as a bot with `min_rating`. I will add a comment.